tests/core/test_from_to_format_conversions.py

```diff
@@ -83,6 +104,16 @@ class Config:
         to_format = "pickle"
 
 
+def custom_to_pickle(data: Any, *args, **kwargs):
+    return data.to_pickle(*args, **kwargs)
+
+
+class OutSchemaPickleCallable(OutSchema):
+    class Config:
+        to_format = custom_to_pickle
+        to_format_buffer = io.BytesIO()
```

Comment:
this should be the class `io.BytesIO` and not an instance of this to avoid potential issues with statefulness.

Then in the `to_format` method:

```python
buffer = None if config.to_format_buffer is None else config.to_format_buffer()
```

Basically, `to_format_buffer` is a callable with no args and produces the appropriate buffer.

Reply:
But the to_format_buffer could also be a file path right?

Reply:
Added that solution with docs, let me know if that looks alright!